### ISSUE TITLE
Globus not foolproof

### ISSUE THREAD

**Issue by ianfoster** [globus/globus-cli#5]
The partial commands:

    globus transfer task 
    globus transfer
    globus

all give:

    AttributeError: 'Namespace' object has no attribute 'fun'

**Comment by sirosen** [comment]
The issue here is that partial commands like there are categorized as `MenuCommand`s internally, as opposed to `FuncCommand`s (which map to functions to run).

I think I just need a handler for this that defaults `MenuCommand`s to print help and exit with status 2.

**Comment by sirosen** [comment]
Hm. This may be more subtle than I expected.
On my own machine:
```shell
$ globus transfer
usage: globus transfer [-h] [-F {json,text}] [--version]  ...
globus transfer: error: too few arguments
$ echo $?
2
```

@ianfoster What version of python are you running? `python --version`?
I wonder if argparse semantics have subtly changed between 2 and 3.
I'm running 2.7 locally, and I'm about to check on 3.x

**Comment by ianfoster** [comment]
I am running v3, specifically Python 3.4.4

**Comment by sirosen** [comment]
I'm running `3.4.2` and it works. Which is disappointing, since it means that isn't an obvious explanation.
Looking for other explanations now.

**Comment by corpulentcoffee** [comment]
FWIW, I see the same problem as Ian on Python 3.5.1.
Python 2.7.11 works correctly and shows the help text.

**Comment by sirosen** [comment]
What I've deduced thusfar is that argument parsing is _succeeding_ for you when it shouldn't.
You're giving argparse a command which has (in argparse-speak) subparsers attached to it, but not supplying a subcommand to match one of those subparsers.

The fact that this is a usage error is something that was broken and then fixed during Python 3.3 development: http://bugs.python.org/issue9253#msg186387

I'm going to fix this by manually setting the `required` attribute on the subparser object, which should be compatible with all relevant pythons, and fix it for the pythons where it is "incorrect" per argparse documentation.
